pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -150,5 +150,12 @@ func Pod(source *v1alpha1.CatalogSource, name string, image string, labels map[s
 			},
 		},
 	}
+	if len(source.Spec.Secrets) > 0 {
+		pod.Spec.ImagePullSecrets = []v1.LocalObjectReference{
```

Comment:
I'd like to see what it looks like if we bind the secrets to the pod's service account rather than the pod itself: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account

Reply:
From what I can see from the doc, if we were to bind image pull secrets to the default SA in the catalog pod namespace all pods using that default SA would get the `imagePullSecrets` field set automatically, regardless of whether they are catalogs or just other random pods in that namespace. Is that accurate? Per https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#verify-imagepullsecrets-was-added-to-pod-spec

Would we use the default SA or a new individual SA for each custom catalog?

Reply:
>I'd like to see what it looks like if we bind the secrets to the pod's service account rather than the pod itself:

Which service account do you mean exactly by the `pod's service account`? This is what I can see is created when I create a new CatalogSource: 

```
$ kubectl get pods -n olm 
NAME                                                READY   STATUS    RESTARTS   AGE
catalog-operator-7fb7fb69d9-6r8hf       1/1        Running             0          25h
my-catalog-sqbf2                                  1/1        Running             0          25h
olm-operator-69f46ff69d-5kvg7            1/1        Running             0          25h
operatorhubio-catalog-9wb2x               1/1        Running             0          25h
packageserver-59dc8f9447-v4xpn       1/1        Running             0          25h 

$ k get serviceaccount -n olm       
NAME                                            SECRETS   AGE
default                                                  1            24h
olm-operator-serviceaccount               1            24h

$ k get service -n olm 
NAME                                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)     AGE
my-catalog                      ClusterIP     10.96.199.188         <none>        50051/TCP    25h
operatorhubio-catalog    ClusterIP     10.96.141.232          <none>        50051/TCP    25h
packageserver-service   ClusterIP     10.96.95.185            <none>        5443/TCP      25h
```